File: app/src/main/java/com/irishinterest/irishinterest/network/api/irishInterest/NotificationOnChange.java
package com.irishinterest.irishinterest.network.api.irishInterest;

import com.irishinterest.irishinterest.model.provider.Provider;
import com.irishinterest.irishinterest.network.api.irishInterest.observer.Data;

public interface NotificationOnChange<T extends Data> {
  void add(T data);
  void modify(T data);
  void delete(T data);
  void updateAll(T data);
  void registerProvider(Provider<T> provider);
}

File: app/src/main/java/com/irishinterest/irishinterest/network/api/irishInterest/NotificationOnChange.kt
package com.irishinterest.irishinterest.network.api.irishInterest

import com.irishinterest.irishinterest.model.provider.Provider
import com.irishinterest.irishinterest.network.api.irishInterest.observer.Data

interface NotificationOnChange<T : Data?> {
    fun add(data: T)
    fun modify(data: T)
    fun delete(data: T)
    fun updateAll(data: T)
    fun registerProvider(provider: Provider<T>?)
}